### PR TITLE
Mmann1123 ml predict

### PR DESCRIPTION
**Pull request by mmann1123**
**Problems**
- requiring passing Featurizer() to geowombat pipline was confusing to users
- no 'predict' method available thus no meaningful model tuning and cv where possible
- inaccurate documentation

**Changes**
- checked for and insert Featurizer into pipeline by default (updated docs)
-  added predict method 
- Updated documentation

**Questions**
- I am using the following code to do the predict back to an xarray object.  I am worried that the `return` line isn't the best way to insert the data back in as a band. Is this how you would do it? 
 
``` python
Y = (
            clf.predict(X)
            .unstack(targ_dim_name)
            .assign_coords(coords={"band": targ_name})
            .expand_dims(dim="band")
            .transpose("time", "band", "y", "x")
        )

        return xr.concat([data, Y.astype(uint16)], dim="band").sel(band=targ_name)
```

- I am working on a way to use missing values so this is being left unresolved here.  I am thinking however that the user could pass a missing value argument.   
     - Would it be better to recast the missing value into the dtype of the array? Or better to recast the array to match the missing value dtype provided by the user?   

``` python
        # no point unit doesn't have nan
        # Y = self._mask_nodata(Y, data, data.attrs["nodatavals"][0])
```

### CODE REVIEW

**Comment by mmann1123** [comment]
**Update**

I implemented nodatavalue masking for the prediction later. I decided to use bumpy array rather than the carry method used elsewhere, seemed faster and little memory overhead.

File: geowombat/ml/classifiers.py [hunk 1]
```diff
@@ -9,6 +9,9 @@
 from sklearn.preprocessing import LabelEncoder
 from sklearn_xarray import wrap, Target
 from sklearn_xarray.model_selection import CrossValidatorWrapper
+from numpy import uint16, nan
```
Comment (jgrss) [inline geowombat/ml/classifiers.py:12]: I would prefer `import numpy as np` but just a minor comment.

**Review by jgrss — APPROVED** [review]
Looks good. Thanks for adding tests with these changes.

**Comment by jgrss** [comment]
> @jgrss Just wondering if you can take a quick look at this. Also would like to keep up with numbered versions and want to make sure I do it right.

We could really use some auto CI/CD semantic versioning, but I don't much time to set that up at the moment.

**Comment by mmann1123** [comment]
>
